ImputationPipeline/Validation/ValidateScoring.wdl

```diff
@@ -0,0 +1,264 @@
+version 1.0
+
+import "../ScoringPart.wdl" as Scoring
+import "https://raw.githubusercontent.com/broadinstitute/palantir-workflows/main/ImputationPipeline/ScoringPart.wdl" as ScoringMain
+
+workflow ValidateScoring {
+	input {
+		File? validationArrays
+		File validationArraysMain
+		File validationWgs
+
+		String population_basename
+
+		File population_loadings
+		File population_meansd
+		File population_pcs
+		File pruning_sites_for_pca # and the sites used for PCA
+		File population_vcf
+
+		File weights
+		File sample_name_map
+		String branch
+
+		Int wgs_vcf_to_plink_mem = 8
+	}
+
+
+
+	call ExtractIDs as extractImputedIDs {
+		input:
+			vcf = select_first([validationArrays, validationArraysMain]),
+			output_basename = "imputed"
+	}
+
+	call ExtractIDs as extractImputedIDsMain {
+		input:
+			vcf = validationArraysMain,
+			output_basename = "imputed"
+	}
+
+
+	call SubsetWeights {
+		input:
+			sites = extractImputedIDs.ids,
+			weights = weights
+	}
+
+	call SubsetWeights as SubsetWeightsMain {
+		input:
+			sites = extractImputedIDsMain.ids,
+			weights = weights
+	}
+
+
+	call Scoring.ScoringImputedDataset as ScoreImputed {
+		input:
+			weights = weights,
+			imputed_array_vcf = select_first([validationArrays, validationArraysMain]),
+			population_basename = population_basename,
+			basename = "imputed",
+			population_loadings = population_loadings,
+			population_meansd = population_meansd,
+			population_pcs = population_pcs,
+			pruning_sites_for_pca = pruning_sites_for_pca,
+			population_vcf = population_vcf,
+			redoPCA = true
+	}
+
+	call ScoringMain.ScoringImputedDataset as ScoreImputedMain {
+		input:
+			weights = weights,
+			imputed_array_vcf = validationArraysMain,
+			population_basename = population_basename,
+			basename = "imputed",
+			population_loadings = population_loadings,
+			population_meansd = population_meansd,
+			population_pcs = population_pcs,
+			pruning_sites_for_pca = pruning_sites_for_pca,
+			population_vcf = population_vcf,
+			redoPCA = true
+	}
+
+	call QCSites {
+		input:
+			input_vcf = validationWgs,
+			output_vcf_basename = "wgsValidation"
+	}
+
+	call Scoring.ScoringImputedDataset as ScoreWGS {
```

Comment:
It would be helpful to have some comments throughout both these WDLs; here, for example, to make clear what the difference between ScoringImputedMain and ScoringWGS is (since it's not just 2 branches).

Highest priority would be comments explaining what the inputs are.

Reply:
good call, I will add some comments